app/src/main/java/com/codingblocks/cbonlineapp/mycourse/MyCourseViewModel.kt

```diff
@@ -144,6 +149,19 @@ class MyCourseViewModel(
             .enqueue(request)
     }
 
+    fun getLeaderboard(crUid: String?) {
+        runIO {
+            when (val response = crUid?.let { repo.fetchLeaderboard(it) }) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    this.enqueue(retrofitCallback { _, response ->
```

Comment:
can you explain what exactly are you trying to achieve here ?

Reply:
This is a network request right.. so am executing the request asyncly with enqueue method with the read callback and thus updating the `leaderboard` with the response received.

Reply:
Or do you suggest `awaitResponse().body()` instead of enquing the request?